Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -107,6 +107,7 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 
         if let _ = entity.id, entity.exists {
             // update
+            try type(of: entity).willUpdate(entity: entity)
```

Comment:
That's probably true. However `E.willUpdate` is arguably more concise and readable vs. `type(of: entity).willUpdate`. I think that's what we use elsewhere in the code as well.

I can do a quick update.